ehrapy/api/preprocessing/encoding/_encode.py

```diff
@@ -504,6 +594,99 @@ def _undo_encoding(
             layers={"original": new_x.copy()},
         )
 
+    @staticmethod
+    def _delete_all_encodings(adata: AnnData) -> Tuple[Optional[np.ndarray], Optional[list]]:
+        """Delete all encoded columns and keep track of their indices
+        Args:
+            adata: The AnnData object to operate on
+
+        Returns:
+            A temporary X were all encoded columns are deleted and all var_names of unencoded columns
+        """
+        var_names = list(adata.var_names)
+        if adata.X is not None and var_names is not None:
+            idx = 0
```

Comment:
This whole implementation has the assumption that the categoricals are after one another. Is this assumption violated at any time? What if we combine several AnnData objects? Joining, merging etc?

Reply:
Currently I can't think of any case where this might be violated, since encode entorces it this way. 

If we merge two AnnData objects, this has to be kept in mind (although we will need custom solutions here anyways since we are likely working with AnnData objects of different shapes)

Reply:
> If we merge two AnnData objects, this has to be kept in mind (although we will need custom solutions here anyways since we are likely working with AnnData objects of different shapes)

People are implementing this for AnnData of course. Some functionality already exists for this.